pandas/tests/frame/test_asof.py

```diff
@@ -106,3 +106,20 @@ def test_all_nans(self):
         result = DataFrame(np.nan, index=[1, 2], columns=['A', 'B']).asof(3)
         expected = Series(np.nan, index=['A', 'B'], name=3)
         tm.assert_series_equal(result, expected)
+
+    # Testing awareness of DataFrame index considering different
+    # UTC and different timezone
+    @pytest.mark.parametrize("stamp,expected",
+              [(Timestamp('2018-01-01 23:22:43.325+00:00'),
+                Series(2.0,name=Timestamp('2018-01-01 23:22:43.325+00:00'))),
+               (Timestamp('2018-01-01 22:33:20.682+01:00'),
```

Comment:
can you add a NaT in here as well (so another case)

Reply:
I don't understand what do you mean. Where exactly I should add a NaT?

Reply:
equiv of the naive case, but with a non-naive Timestamp (we may already have this test, if so, just add on to it)
```
In [2]: df = DataFrame(data=[1, 2],
   ...:                       index=[pd.Timestamp('2018-01-01 21:00:05.001'),
   ...:                             pd.Timestamp('2018-01-01 22:35:10.550')])
   ...:                             

In [3]: df
Out[3]: 
                         0
2018-01-01 21:00:05.001  1
2018-01-01 22:35:10.550  2

In [4]: df.asof('2018-01-01 22:35:10.550')
Out[4]: 
0    2.0
Name: 2018-01-01 22:35:10.550000, dtype: float64

In [5]: df.asof(pd.NaT)
Out[5]: 
0    1.0
Name: NaT, dtype: float64

In [7]: df.asof('2018-01-01')
Out[7]: 
0   NaN
Name: 2018-01-01 00:00:00, dtype: float64

```

Reply:
thanks for the clarification. I will check it out.

Reply:
with the fix `where.values` in asof_loc I have for the NaT case the following:

    df
    Out[18]: 
                             0
    2018-01-01 21:00:05.001  1
    2018-01-01 22:35:10.550  2
    
    result=df.asof(pd.NaT)

    /Users/.../pandas/core/indexes/base.py:2382: 
    FutureWarning: In the future, 'NAT < x' and 'x < NAT' will always be False.
    result[(locs == 0) & (where.values < self.values[first])] = -1

    result
    Out[20]: 
    0   NaN
    Name: NaT, dtype: float64
    
Is 1 really the desired value in this situation?